docs/book/src/sway-program-types/libraries.md

```diff
@@ -86,20 +85,104 @@ The `dep` keyword in the main library includes a dependency on another library,
 
 ## Using Libraries
 
-Libraries can be imported using the `use` keyword and with a `::` separating the name of the library and the import.
+There are two types of libraries categorised based on where it is located and how we can import them.
 
-Here is an example of importing the `get<T>` and `store<T>` functions from the `storage` library.
+### Internal Libraries
 
-```sway
-use std::storage::{get, store};
+Internal libraries are located within the main `src` directory with the main program files as shown below:
+
+```bash
+$ tree
+.
+├── Cargo.toml
+├── Forc.toml
+└── src
+    ├── internal_lib.sw
+    └── my_library.sw
 ```
 
-Wildcard imports using `*` are supported, but it is always recommended to use explicit imports where possible.
 
-Libraries _other than the standard library_ have to be added as a dependency in `Forc.toml`. This can be done by adding a path to the library in the `[dependencies]` section. For example:
+As `internal_lib` is an internal library, it can be imported to use inside `my_library` as follows:
+- Use the `dep` keyword followed by the library name to make the internal library a dependancy
+- Use the `use` keyword with a `::` separating the name of the library and the import
+
+```sway
+library my_library;
+
+dep internal_lib;
+
+use internal_lib::mint;
+```
+
+### External Libraries
+
+External libraries are located outside the main `src` directory as shown below:
+
+```bash
+$ tree
+.
+├── my_library
+│   ├── Cargo.toml
+│   ├── Forc.toml
+│   └─── src
+│       └── lib.sw
+│
+└── external_lib
+    ├── Cargo.toml
+    ├── Forc.toml
+    └─── src
+        └── lib.sw
+```
+        
+As `external_lib` is outside the main `src` directory of `my_library`, it needs to be added as a dependancy in `Forc.toml` of `my_library` by adding the library path in the `dependancies` section as below in order to be imported:
```

Comment:
```suggestion
As `external_lib` is outside the `src` directory of `my_project`, it needs to be added as a dependency in the `Forc.toml` file of `my_project`, by adding the library path in the `dependencies` section as shown below, before it can be imported:
```